scipy/optimize/minpack.py

```diff
@@ -452,9 +445,16 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
                 cov_x = inv(dot(transpose(R), R))
             except (LinAlgError, ValueError):
                 pass
-        return (retval[0], cov_x) + retval[1:-1] + (mesg, info)
+        return (retval[0], cov_x) + retval[1:-1] + (errors[info][0], info)
     else:
-        return (retval[0], info)
+        if info not in LEASTSQ_SUCCESS:
+            if info > 0:
+                warnings.warn(errors[info][0], RuntimeWarning)
+            elif info == 0:
+                raise errors[info][1](errors[info][0])
+            else:
+                raise errors['unknown'][1](errors['unknown'][0])
```

Comment:
If I understand the code in __minpack.h correctly `info` in this module/function can never be negative (see https://github.com/scipy/scipy/blob/11509c4a98edded6c59423ac44ca1b7f28fba1fd/scipy/optimize/__minpack.h#L597 and https://github.com/scipy/scipy/blob/11509c4a98edded6c59423ac44ca1b7f28fba1fd/scipy/optimize/__minpack.h#L499). Is this correct?